src/main.cpp

```diff
@@ -0,0 +1,210 @@
+}
+
+void playMenu() {
+  printHeader("SELECT MODE");
+  printItem("1. Player vs Player");
+  printItem("2. Player vs CPU");
+  printItem("3. Return to Main Menu");
+  printBorder();
+  printf("Select option: ");
+  int option = getValidatedInput([](int input) { 
+    return input >= 1 && input <= 3;
+  });
+
+  switch (option) {
+    case 1:
+      board.setCpu(isCpu = false);
+      selectBoardSize();
+    case 2:
+      cpuInit();
+      board.setCpu(isCpu = true);
+      selectBoardSize();
+    case 3:
+      return;
+    default:
+      printf("Error: Invalid option\n");
+  }
+}
+
+void selectBoardSize() {
+  printHeader("SELECT BOARD SIZE");
+  printItem("1. 5");
+  printItem("2. 7");
+  printItem("3. 9");
+  printItem("4. Return to Main Menu");
+  printBorder();
+  printf("Select option: ");
+  int option = getValidatedInput([](int input) { 
+    return input >= 1 && input <= 4; 
+  });
+
+  // Build board and play game
+  if (option != 4) {
+    board.reset((option * 2) + 3); // Converts option 1,2,3 to size 5,7,9
+    playGame();
+  }
+}
+
+void playGame() {
+  int p1Move = 0;
+  int p2Move = 0;
+  int p1LastMove = 0;
+
+  while (board.checkWinner() == -1) {
+    // P1
+#ifdef DEBUG
+    board.printBoard(P1);
+#else
+    board.printPlayerMoves(P1);
+#endif
+    printf("Player 1, select a number: ");
+    p1Move = getValidatedInput([](int input) { 
+      return board.isValidMove(P1, input); 
+    });
+
+    // P2 or CPU
+    if (isCpu) {
+      // CPU
+      do {
+        p2Move = cpuMove(board.p2, p1LastMove, board.boardSize);
+      } while (!board.isValidMove(P2, p2Move));
+    } else {
+      // P2
+      printHeader("NUMBERS ROYALE");
+      printf("Pass the device to Player 2\n");
+      waitForEnter();
+
+      board.printPlayerMoves(P2);
+      printf("Player 2, select a number: ");
+      p2Move = getValidatedInput([](int input) { 
+        return board.isValidMove(P2, input); 
+      });
+    }
+
+    // Play the round
+    board.playMove(p1Move, p2Move);
+    p1LastMove = p1Move;
+
+    // Round results
+    if (p1Move == p2Move) {
+      printHeader("It's a tie!");
+    } else if (p1Move > p2Move) {
+      printHeader("Player 1 wins the round!");
+    } else if (isCpu) {
+      printHeader("CPU wins the round!");
+    } else {
+      printHeader("Player 2 wins the round!");
+    }
+    std::string p1MoveText = "Player 1 move: " + std::to_string(p1Move);
+    std::string p2MoveText = isCpu ? "CPU move: " : "Player 2 selected: ";
+    p2MoveText += std::to_string(p2Move);
+    printItem(p1MoveText.c_str());
+    printItem(p2MoveText.c_str());
+    board.printScore();
+
+    waitForEnter(); 
+  }
+
+  // Game is over
+}
+
+int main() {
+  int option;
+  // Main loop
+  while (true) {
+    option = mainMenu();
+    if (option == 3) {
+      break;
+    }
+
+    switch (option) {
+      case 1:
+        playMenu();
+        break;
+      case 2:
+        howToPlay();
+        break;
+      default: 
```

Comment:
i love magic numbers

Reply:
grr xD, I should probably make it an enum. but since enums are 0-index, would it be ok to make a dummy option 1?

e.g.
```
enum Menu {
  DUMMY,
  OPTION_1,
  OPTION_2
}
```

Reply:
ew you don't know cpp cringe

Reply:
```cpp
enum Menu {
  MENU = 1,
  HOW_TO_PLAY
};
```
you can explicitly set each of them, otherwise it starts incrementing by 1